File: examples/hook-template/python/command_hook.py
#!/usr/bin/env python3
import json
import sys


def main() -> int:
    raw = sys.stdin.read()
    payload = json.loads(raw) if raw.strip() else {}
    hook_context = payload.get("_hook_context") or {}
    summary = {
        "ok": True,
        "language": "python",
        "event_type": payload.get("event_type", ""),
        "run_id": payload.get("run_id", ""),
        "phase": hook_context.get("phase", ""),
        "message": "Handled %s for run %s"
        % (payload.get("event_type", "unknown"), payload.get("run_id", "unknown")),
    }
    json.dump(summary, sys.stdout, ensure_ascii=True, indent=2)
    sys.stdout.write("\n")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: examples/hook-template/node/command_hook.mjs
#!/usr/bin/env node

const chunks = [];

process.stdin.on('data', (chunk) => {
  chunks.push(chunk);
});

process.stdin.on('end', () => {
  const raw = Buffer.concat(chunks).toString('utf8').trim();
  const payload = raw ? JSON.parse(raw) : {};
  const hookContext = payload._hook_context || {};
  const summary = {
    ok: true,
    language: 'node',
    event_type: payload.event_type || '',
    run_id: payload.run_id || '',
    phase: hookContext.phase || '',
    message: `Handled ${payload.event_type || 'unknown'} for run ${payload.run_id || 'unknown'}`,
  };
  process.stdout.write(`${JSON.stringify(summary, null, 2)}\n`);
});

process.stdin.resume();
